File: plugin-widgets/platforms/android/java/com/akylas/weather/widgets/WidgetsLogger.kt
package com.akylas.weather.widgets

import android.util.Log

/**
 * Simple centralized logger for widgets package. Uses Android Log.
 * Logging default follows BuildConfig.DEBUG; toggle WidgetsLogger.enabled if you need runtime change.
 */
object WidgetsLogger {
    private const val PREFIX = "Widgets"

    // Default to debug builds
    var enabled: Boolean  = true

    fun d(tag: String, message: String) {
        if (!enabled) return
        Log.d("$PREFIX:$tag", message)
    }

    fun i(tag: String, message: String) {
        if (!enabled) return
        Log.i("$PREFIX:$tag", message)
    }

    fun w(tag: String, message: String, t: Throwable? = null) {
        if (!enabled) return
        if (t == null) Log.w("$PREFIX:$tag", message) else Log.w("$PREFIX:$tag", message, t)
    }

    fun e(tag: String, message: String, t: Throwable? = null) {
        if (!enabled) return
        if (t == null) Log.e("$PREFIX:$tag", message) else Log.e("$PREFIX:$tag", message, t)
    }
}

File: plugin-widgets/platforms/ios/src/WidgetsLogger.swift
import Foundation
import os.log

/**
 * Simple centralized logger for widgets package. Uses os_log.
 * Logging can be toggled via WidgetsLogger.enabled property.
 */
struct WidgetsLogger {
    // Default to enabled
    static var enabled: Bool = true
    
    static func d(_ tag: String, _ message: String) {
        guard enabled else { return }
        print("CONSOLE DEBUG: [\(tag)] \(message)")
        // os_log("[%@] %@", log: .default, type: .debug, tag, message)
    }
    
    static func i(_ tag: String, _ message: String) {
        // guard enabled else { return }
        print("CONSOLE LOG: [\(tag)] \(message)")
        // os_log("[%@] %@", log: .default, type: .info, tag, message)
    }
    
    static func w(_ tag: String, _ message: String) {
        // guard enabled else { return }
        print("CONSOLE WARN: [\(tag)] \(message)")
        // os_log("[%@] %@", log: .default, type: .default, tag, message)
    }
    
    static func e(_ tag: String, _ message: String) {
        // guard enabled else { return }
        print("CONSOLE ERROR: [\(tag)] \(message)")
        // os_log("[%@] %@", log: .default, type: .error, tag, message)
    }
}
